### PR TITLE
Update docs for azd

### PR DESCRIPTION
**Pull request by billwert**
Add new developer credential to README and fixup mermaid.

### CODE REVIEW

File: sdk/identity/azure-identity/README.md [hunk 1]
```diff
@@ -430,6 +429,8 @@ Not all credentials require this configuration. Credentials that authenticate th
   </tbody>
 </table>
 
+> __Note:__ `AzureDeveloperCliCredential` is in beta and its name may change before GA release.
```
Comment (scottaddie) [inline sdk/identity/azure-identity/README.md:432]:
```suggestion
> __Note:__ `AzureDeveloperCliCredential` is in beta and its name may change before the stable release.
```

File: sdk/identity/azure-identity/images/mermaidjs/DefaultAzureCredentialAuthFlow.md [hunk 2]
```diff
@@ -22,7 +22,7 @@ flowchart LR;
     %% Add API ref links to credential type boxes
     click A "https://docs.microsoft.com/java/api/com.azure.identity.environmentcredential?view=azure-java-stable" _blank;
     click B "https://docs.microsoft.com/java/api/com.azure.identity.managedidentitycredential?view=azure-java-stable" _blank;
-    click C "https://docs.microsoft.com/java/api/com.azure.identity.intellijcredential?view=azure-java-stable" _blank;
+    click D "https://docs.microsoft.com/java/api/com.azure.identity.intellijcredential?view=azure-java-stable" _blank;
```
Comment (scottaddie) [inline sdk/identity/azure-identity/images/mermaidjs/DefaultAzureCredentialAuthFlow.md:25]: While you're in here, let's replace "docs.microsoft.com" with "learn.microsoft.com" for each of these links.

**Review by scottaddie — COMMENTED** [review]
In the mermaid Markdown file, also change `linkStyle 4 stroke-width:0px;` to `linkStyle 5 stroke-width:0px;`

File: sdk/identity/azure-identity/images/mermaidjs/DefaultAzureCredentialAuthFlow.md (hunk 2 above)
Comment (billwert) [reply sdk/identity/azure-identity/images/mermaidjs/DefaultAzureCredentialAuthFlow.md]: Oh, good catch.

File: sdk/identity/azure-identity/images/mermaidjs/DefaultAzureCredentialAuthFlow.md [hunk 3]
```diff
@@ -5,24 +5,24 @@
 %% 2. Run command: mmdc -i DefaultAzureCredentialAuthFlow.md -o DefaultAzureCredentialAuthFlow.svg
 
 flowchart LR;
```
Comment (scottaddie) [inline sdk/identity/azure-identity/images/mermaidjs/DefaultAzureCredentialAuthFlow.md:7]: Nit: Please install v8.14.0 of the Mermaid CLI and then regenerate the SVG. It fixes this issue: 
image

And while you're in here, it probably makes sense to update the Markdown comment to say v8.14.0 was used to generate the SVG.

**Review by scottaddie — APPROVED** [review]

**Review by scottaddie — APPROVED** [review]

**Pull request merged**